cranelift/codegen/src/isa/aarch64/inst/emit.rs

```diff
@@ -1337,6 +1337,31 @@ impl MachInstEmit for Inst {
                     }
                 }
             }
+            &Inst::MovPReg { rd, rm } => {
+                let rd = allocs.next_writable(rd);
+                let rm: Reg = rm.into();
+                assert!(rd.to_reg().class() == rm.class());
+                assert!(rm.class() == RegClass::Int);
+
+                // MOV to SP is interpreted as MOV to XZR instead. And our codegen
+                // should never MOV to XZR.
+                assert!(rd.to_reg() != stack_reg());
+
+                if rm == stack_reg() {
```

Comment:
Rather than duplicate the `mov` code, could we do `Inst::Mov { .. }.emit(...)`, in the same way other pseudoinsts work?